Source/Core/VideoCommon/NetPlayChatUI.h

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <array>
+#include <deque>
+#include <functional>
+#include <memory>
+#include <string>
+#include <utility>
+
+struct ImVec4;
+
+class NetPlayChatUI
+{
+public:
+  explicit NetPlayChatUI(std::function<void(const std::string&)> callback);
+  ~NetPlayChatUI() = default;
+
+  void Display();
+  void AppendChat(const std::string& message, const std::string& color);
+  void SendMessage();
+
+private:
+  char m_message_buf[256] = {};
+  bool m_scroll_to_bottom = false;
+  bool m_is_scrolled_to_bottom = true;
+
+  std::deque<std::pair<std::string, std::array<float, 3>>> m_messages;
```

Comment:
```suggestion
  std::deque<std::pair<std::string, ImVec4>> m_messages;
```
To actually use the struct you forward declared.

Reply:
Are we sure this properly works without fully knowing the struct? Pointers should be fine (since they're always the same size), but this should allocate sufficiant space for storage; I'm not too sure this is ok.

Reply:
It doesn't. The forward declaration is actually an unintended left-over.

Reply:
I'm going to blame that I've been working with smart-pointers too much lately.